proposals/field-keyword.md

```diff
@@ -230,74 +230,110 @@ public struct S
 
-```cs
-class C
-{
-    // ⚠️ CS8618: Non-nullable property 'P' must contain a
-    // non-null value when exiting constructor.
-    public string P { get => field; set => field = value; }
-}
-```
+Goals:
 
-No warning should be produced if the property is initialized to a non-null value via constructor assignment or property initializer:
+- A reasonable level of null-safety should be ensured for various usage patterns of the `field` keyword feature.
 
-```cs
-class C
-{
-    public C() { P = ""; }
+- Patterns that use the `field` keyword should feel as though they've always been part of the language. Avoid making the user jump through hoops to enable Nullable Reference Types in code that is perfectly idiomatic for the `field` keyword feature.
 
-    public string P { get => field; set => field = value; }
-}
-```
+One of the key scenarios is lazily initialized properties:
 
 ```cs
-class C
+public class C
 {
-    public string P { get => field; set => field = value; } = "";
+    public C() { } // It would be undesirable to warn about 'Prop' being uninitialized here
+
+    string Prop => field ??= GetPropValue();
 }
 ```
 
-#### Open question: nullability of `field`
+The following nullability rules will apply not just to properties that use the `field` keyword, but also to existing auto properties.
 
-In the same vein as how `var` infers as nullable for reference types, the `field` type is the nullable type of the property whenever the property's type is not a value type. Otherwise, `field ??` would appear to be followed by dead code, and it avoids producing a misleading warning in the following example:
+#### Terms
 
+A property is a *field-backed property* if it meets any of the following conditions:
+1. It contains any automatically implemented accessors (`get;`/`set;`/`init;`).
+2. It uses the `field` keyword within any accessor bodies.
+
+Some examples of *field-backed properties* include:
 ```cs
-public string AmbientValue
-{
-    get => field ?? parent.AmbientValue;
-    set
-    {
-        if (value == parent.AmbientValue)
-            field = null; // No warning here. Resume following the parent's value.
-        else
-            field = value; // Stop following the parent's value
-    }
-}
+public string Prop1 { get; set; }
+public string Prop2 { get => field; set; }
+public string? Prop3 { get => field; }
 ```
 
-`var` was designed to declare nullability so that subsequent assignments to the variable could be nullable, due to established patterns in C#. It's expected that the same rationale would apply to property backing fields.
+The variable denoted by the `field` keyword in a property's accessors is the *backing field* of that property.
```

Comment:
Should I move this up into the Glossary section at the top of the file?

Reply:
Yeah feel free to do this and eliminate anything redundant.

Reply:
Done.